packages/components/src/components/dataDisplay/Badge/Badge.stories.tsx

```diff
@@ -58,10 +77,19 @@ const AllTemplate = () => {
 
 export const All = AllTemplate.bind({});
 
-const Template = (args) => <Badge {...args}></Badge>;
+const Template = ({ showLeadingICon, showTrailingIcon, ...args }) => (
+  <Badge
+    {...(showLeadingICon && { leadingIcon: <Icon /> })}
+    {...(showTrailingIcon && { trailingIcon: <Icon /> })}
```

Comment:
I'm not sure `<Icon />` is actually rendering an icon based on the storybook generated. Could you try using a different icon?

Reply:
Fixed!